engines/zvision/detection.cpp

```diff
@@ -120,6 +125,131 @@ bool ZVision::ZVision::canSaveGameStateCurrently() {
 }
 
+Common::KeymapArray ZVisionMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace ZVision;
+
+	Keymap *mainKeymap = new Keymap(Keymap::kKeymapTypeGame, mainKeymapId, "Z-Vision");
+
+	Action *act;
+
+	act = new Action("LCLK", _("Left Click"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	mainKeymap->addAction(act);
+
+	act = new Action("RCLK", _("Right Click"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionMoveUp, _("Look Up"));
+	act->setCustomEngineActionEvent(kZVisionActionUp);
+	act->addDefaultInputMapping("UP");
+	act->addDefaultInputMapping("JOY_UP");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionMoveDown, _("Look Down"));
+	act->setCustomEngineActionEvent(kZVisionActionDown);
+	act->addDefaultInputMapping("DOWN");
+	act->addDefaultInputMapping("JOY_DOWN");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionMoveLeft, _("Turn Left"));
+	act->setCustomEngineActionEvent(kZVisionActionLeft);
+	act->addDefaultInputMapping("LEFT");
+	act->addDefaultInputMapping("JOY_LEFT");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionMoveRight, _("Turn Right"));
+	act->setCustomEngineActionEvent(kZVisionActionRight);
+	act->addDefaultInputMapping("RIGHT");
+	act->addDefaultInputMapping("JOY_RIGHT");
+	mainKeymap->addAction(act);
+
+	act = new Action("FPS", _("Show FPS"));
+	act->setCustomEngineActionEvent(kZVisionActionShowFPS);
+	act->addDefaultInputMapping("F10");
+	mainKeymap->addAction(act);
+
+	act = new Action("HELP", _("Help"));
+	act->setKeyEvent(KEYCODE_F1);
+	act->addDefaultInputMapping("F1");
+	act->addDefaultInputMapping("JOY_LEFT_TRIGGER");
+	mainKeymap->addAction(act);
+
+	act = new Action("INV", _("Inventory"));
+	act->setKeyEvent(KEYCODE_F5);
+	act->addDefaultInputMapping("F5");
+	act->addDefaultInputMapping("JOY_LEFT_SHOULDER");
+	mainKeymap->addAction(act);
+
+	act = new Action("SPELL", _("Spellbook"));
+	act->setKeyEvent(KEYCODE_F6);
+	act->addDefaultInputMapping("F6");
+	act->addDefaultInputMapping("JOY_RIGHT_SHOULDER");
+	mainKeymap->addAction(act);
+
+	act = new Action("SCORE", _("Score"));
+	act->setKeyEvent(KEYCODE_F7);
+	act->addDefaultInputMapping("F7");
+	act->addDefaultInputMapping("JOY_RIGHT_TRIGGER");
+	mainKeymap->addAction(act);
+
+	act = new Action("AWAY", _("Put away object"));
+	act->setKeyEvent(KEYCODE_F8);
+	act->addDefaultInputMapping("F8");
+	act->addDefaultInputMapping("JOY_X");
+	mainKeymap->addAction(act);
+
+	act = new Action("COIN", _("Extract coin"));
+	act->setKeyEvent(KEYCODE_F9);
+	act->addDefaultInputMapping("F9");
+	act->addDefaultInputMapping("JOY_Y");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionSave, _("Save"));
```

Comment:
Any particular reason for not using custom actions for these? (ZVision::shortKeys)

Reply:
I'm not sure how I missed that. Now done, thanks.